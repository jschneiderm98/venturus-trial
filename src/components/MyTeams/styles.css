.Teams {
  box-shadow: 0px 0px 1px 1px #f3ecf2;
  background-color: #ffffff;
  width: 700px;
  height: 500px;
  border-radius: 10px 10px 0 0;
}

.Top {
  height: 80px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  font-size: 24px;
  font-weight: 700;
  border-bottom: 1px solid #f8f1f7;
}

.Top:first-child {
  padding-left: 40px;
  color: #64008b;
}

.Square {
  margin-right: 40px;
  --size: 40px;
  background-image: linear-gradient(rgb(176, 0, 119), rgb(137, 0, 130));
  border-radius: 10px;
  height: var(--size);
  text-align: center;
  width: var(--size);
  display: flex;
  justify-content: center;
  align-items: center;
  box-shadow: 0px 5px 10px 0px rgba(199, 0, 71, 0.6);
}

.Plus {
  font-size: calc(var(--size) / 2);
  content: '';
  color: #f3f4f7;
  font-weight: 700;
}

.Plus::before {
  content: '\FF0B';
}

.ListComponent {
  display: flex;
  justify-content: center;
  align-items: center;
}

table {
  width: 90%;
  margin-top: 20px;
  border-collapse: collapse;
}

thead>tr>th {
  border-bottom: 1em solid transparent;
}

th,
td {
  height: 30px;
  font-weight: 500;
  text-align: start;
  padding-left: 20px;
  color: #333333;
}

.TableHeaderText {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.TableHeaderText:nth-child(1) {
  padding-right: 15px;
}

.arrowWrapper {
  display: flex;
  flex-direction: column;
  font-size: 7px;
}

th:first-child {
  border-right: 1px solid #f8f1f7;
  width: 30%;
}

td {
  padding-top: 20px;
  padding-bottom: 20px;
  border-bottom: 1px solid #f8f1f7;
}

td:first-child {
  border-radius: 15px 0px 0px 15px;
}

td:nth-child(2) {
  border-radius: 0px 15px 15px 0px;
}


tbody>tr:hover td {
  background-color: #f9eaf7;
  color: #93006d;
  border-bottom: 0;
}

.HoverField {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.HoverField>div {
  margin-right: 10px;
}

.IconContainer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.DispHover {
  margin-right: 10px;
  display: none;
}

tbody>tr:hover .DispHover {
  display: block;
}