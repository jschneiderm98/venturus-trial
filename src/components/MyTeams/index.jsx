import React from 'react';
import { MdDelete, MdShare, MdEdit } from 'react-icons/md';
import './styles.css';

export default function MyTeams() {
  const itens = [
    { name: 'Barcelona', description: 'Barcelona squad', display: false },
    { name: 'Real Madrid', description: 'Real Madrid squad', display: false },
    { name: 'Milan', description: 'Milan squad', display: false },
  ];

  return (
    <div className="Teams">
      <div className="Top">
        <span>My teams</span>
        <div className="Square">
          <span className="Plus" />
        </div>
      </div>
      <div className="ListComponent">
        <table>
          <thead>
            <tr>
              <th>
                <div className="TableHeaderText">
                  <span>Name</span>
                  <div className="arrowWrapper">
                    <span>&#9650;</span>
                    <span>&#9660;</span>
                  </div>
                </div>
              </th>
              <th>
                <div className="TableHeaderText">
                  <span>Description</span>
                  <div className="arrowWrapper">
                    <span>&#9650;</span>
                    <span>&#9660;</span>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody>
            {itens && itens.map((item) => (
              <tr>
                <td>{item.name}</td>
                <td>
                  <div className="HoverField">
                    <span>{item.description}</span>
                    <div className="IconContainer">
                      <div className="DispHover"><MdDelete /></div>
                      <div className="DispHover"><MdShare /></div>
                      <div className="DispHover"><MdEdit /></div>
                    </div>
                  </div>
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    </div>
  );
}
