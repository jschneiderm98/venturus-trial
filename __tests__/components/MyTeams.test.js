import React from 'react';
import { render } from '@testing-library/react';
import MyTeams from '../../src/components/MyTeams';

describe('Testing MyTeams component', () => {
  test('Should render', () => {
    const tree = render(<MyTeams />);

    expect(tree).toMatchSnapshot();
  });
});
