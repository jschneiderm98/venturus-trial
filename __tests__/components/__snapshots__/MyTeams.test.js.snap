// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MyTeams component Should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Teams"
      >
        <div
          class="Top"
        >
          <span>
            My teams
          </span>
          <div
            class="Square"
          >
            <span
              class="Plus"
            />
          </div>
        </div>
        <div
          class="ListComponent"
        >
          <table>
            <thead>
              <tr>
                <th>
                  <div
                    class="TableHeaderText"
                  >
                    <span>
                      Name
                    </span>
                    <div
                      class="arrowWrapper"
                    >
                      <span>
                        ▲
                      </span>
                      <span>
                        ▼
                      </span>
                    </div>
                  </div>
                </th>
                <th>
                  <div
                    class="TableHeaderText"
                  >
                    <span>
                      Description
                    </span>
                    <div
                      class="arrowWrapper"
                    >
                      <span>
                        ▲
                      </span>
                      <span>
                        ▼
                      </span>
                    </div>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  Barcelona
                </td>
                <td>
                  <div
                    class="HoverField"
                  >
                    <span>
                      Barcelona squad
                    </span>
                    <div
                      class="IconContainer"
                    >
                      <div
                        class="DispHover"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                      </div>
                      <div
                        class="DispHover"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                          />
                        </svg>
                      </div>
                      <div
                        class="DispHover"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  Real Madrid
                </td>
                <td>
                  <div
                    class="HoverField"
                  >
                    <span>
                      Real Madrid squad
                    </span>
                    <div
                      class="IconContainer"
                    >
                      <div
                        class="DispHover"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                      </div>
                      <div
                        class="DispHover"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                          />
                        </svg>
                      </div>
                      <div
                        class="DispHover"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </td>
              </tr>
              <tr>
                <td>
                  Milan
                </td>
                <td>
                  <div
                    class="HoverField"
                  >
                    <span>
                      Milan squad
                    </span>
                    <div
                      class="IconContainer"
                    >
                      <div
                        class="DispHover"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                      </div>
                      <div
                        class="DispHover"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                          />
                        </svg>
                      </div>
                      <div
                        class="DispHover"
                      >
                        <svg
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Teams"
    >
      <div
        class="Top"
      >
        <span>
          My teams
        </span>
        <div
          class="Square"
        >
          <span
            class="Plus"
          />
        </div>
      </div>
      <div
        class="ListComponent"
      >
        <table>
          <thead>
            <tr>
              <th>
                <div
                  class="TableHeaderText"
                >
                  <span>
                    Name
                  </span>
                  <div
                    class="arrowWrapper"
                  >
                    <span>
                      ▲
                    </span>
                    <span>
                      ▼
                    </span>
                  </div>
                </div>
              </th>
              <th>
                <div
                  class="TableHeaderText"
                >
                  <span>
                    Description
                  </span>
                  <div
                    class="arrowWrapper"
                  >
                    <span>
                      ▲
                    </span>
                    <span>
                      ▼
                    </span>
                  </div>
                </div>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                Barcelona
              </td>
              <td>
                <div
                  class="HoverField"
                >
                  <span>
                    Barcelona squad
                  </span>
                  <div
                    class="IconContainer"
                  >
                    <div
                      class="DispHover"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </div>
                    <div
                      class="DispHover"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                        />
                      </svg>
                    </div>
                    <div
                      class="DispHover"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                Real Madrid
              </td>
              <td>
                <div
                  class="HoverField"
                >
                  <span>
                    Real Madrid squad
                  </span>
                  <div
                    class="IconContainer"
                  >
                    <div
                      class="DispHover"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </div>
                    <div
                      class="DispHover"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                        />
                      </svg>
                    </div>
                    <div
                      class="DispHover"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </td>
            </tr>
            <tr>
              <td>
                Milan
              </td>
              <td>
                <div
                  class="HoverField"
                >
                  <span>
                    Milan squad
                  </span>
                  <div
                    class="IconContainer"
                  >
                    <div
                      class="DispHover"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                    </div>
                    <div
                      class="DispHover"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                        />
                      </svg>
                    </div>
                    <div
                      class="DispHover"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
